src/main/java/org/rumbledb/expressions/control/TypeSwitchExpression.java

```diff
@@ -5,57 +5,56 @@
 import org.rumbledb.expressions.AbstractNodeVisitor;
 import org.rumbledb.expressions.Expression;
 import org.rumbledb.expressions.Node;
-import org.rumbledb.expressions.primary.VariableReferenceExpression;
 
 import java.util.ArrayList;
 import java.util.List;
 
 public class TypeSwitchExpression extends Expression {
 
     private final Expression testCondition;
-    private final List<TypeSwitchCaseExpression> cases;
+    private final List<TypeswitchCase> cases;
     private final Expression defaultExpression;
-    private final VariableReferenceExpression defaultVariableReferenceExpression;
+    private final String defaultVariableName;
```

Comment:
I have an alternative suggestion on how to handle the default case. How about instead of storing the defaultVariableName and defaultExpression as separate fields, we refactored these into a TypeSwitchCase as well? I believe this would make code much cleaner in some places (ie. in visitors)

Reply:
Very good idea. this is done.

Reply:
I'm glad to hear that. Please refer to my other comment for hopefully addressing this overhead as well.